IdentityCore/src/broker_operation/request/interactive_token_request/MSIDBrokerOperationInteractiveTokenRequest.m

```diff
@@ -0,0 +1,104 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000 && !MSID_EXCLUDE_WEBKIT
+#import <AuthenticationServices/ASAuthorizationOpenIDRequest.h>
+#import "MSIDBrokerOperationInteractiveTokenRequest.h"
+#import "MSIDPromptType_Internal.h"
+#import "MSIDBrokerOperationRequestFactory.h"
+#import "MSIDPromptType_Internal.h"
+#import "MSIDAccountIdentifier+MSIDJsonSerializable.h"
+#import "MSIDInteractiveRequestParameters.h"
+
+@implementation MSIDBrokerOperationInteractiveTokenRequest
+
++ (void)load
+{
+    if (@available(iOS 13.0, *))
+    {
+        [MSIDBrokerOperationRequestFactory registerOperationRequestClass:self operation:self.operation];
+    }
+}
+
++ (instancetype)tokenRequestWithParameters:(MSIDInteractiveRequestParameters *)parameters
+                                     error:(NSError **)error
+{
+    __auto_type request = [MSIDBrokerOperationInteractiveTokenRequest new];
+    BOOL result = [self fillRequest:request withParameters:parameters error:error];
+    if (!result) return nil;
+    
+    request.accountIdentifier = parameters.accountIdentifier;
+    request.loginHint = parameters.loginHint;
+    request.promptType = parameters.promptType;
+    
+    return request;
+}
+
+#pragma mark - MSIDBrokerOperationRequest
+
++ (NSString *)operation
+{
+    return ASAuthorizationOperationLogin;
+}
+
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+    
+    if (self)
+    {
+        NSError *localError;
+        // We have flat json dictionary, that is why we are passing the whole json to the MSIDAccountIdentifier.
+        _accountIdentifier = [[MSIDAccountIdentifier alloc] initWithJSONDictionary:json error:&localError];
+        if (localError) MSID_LOG_WITH_CORR_PII(MSIDLogLevelWarning, nil, @"Failed to parse MSIDAccountIdentifier %@", MSID_PII_LOG_MASKABLE(localError));
+        
+        _loginHint = [json msidStringObjectForKey:MSID_BROKER_LOGIN_HINT_KEY];
+        
+        NSString *promptString = [json msidStringObjectForKey:MSID_BROKER_PROMPT_KEY];
+        _promptType = MSIDPromptTypeFromString(promptString);
+    }
+    
+    return self;
+}
+
+- (NSDictionary *)jsonDictionary
+{
+    NSMutableDictionary *json = [[super jsonDictionary] mutableCopy];
+    if (!json) return nil;
+    
+    NSDictionary *accountIdentifierJson = [self.accountIdentifier jsonDictionary];
+    if (accountIdentifierJson) [json addEntriesFromDictionary:accountIdentifierJson];
+    
+    json[MSID_BROKER_LOGIN_HINT_KEY] = self.loginHint;
+    
+    NSString *promptString = MSIDPromptParamFromType(self.promptType);
+    if (!promptString) return nil;
```

Comment:
nit: I'd recommend keeping symmetry in required parameters in serialization and deserialization. Here, valid prompt type seems mandatory, but not in initialization.

Reply:
promptString will never be nil, removed this return